### ISSUE TITLE
Understanding TicTacToe performance

### ISSUE THREAD

**Issue by tscode** [roSievers/Jtac.jl#1]
The networks starts learning TicTacToe, but there most certainly are still some issues: even when the loss becomes very small, it does not play optimally. The following are some basic points that we should implement/fix in order to understand the situation better.

- [x] Use symmetry of the game to augment the datasets used for training.
- [ ] Check the development of the weights / implement L1/L2 regularization
- [ ] Check the influence of different other types of regularization (dropout, batchnorm)
- [ ] Way of visualizing the network value and policy evaluations for debugging purposes
- [x] Try different ways of representing the game as data: More channels that separate the players with values in {0, 1} may be more expressive that {-1, 0, 1}? [This did not work well]
- [x] Influence of a lower/higher exploration parameter [Random selfplay-branching works well!]

**Comment by tscode** [comment]
Analyzing the network policies, we realized that there are two primary reasons why the network did not perform flawlessly.
* During the playouts for performance evaluation, we drew actions from the network-policy. However, the network did not completely unlearn stupid moves; it only gave them very small probabilities (~ 1%). This 1%, however, was apparently enough to make the network loose in about 5-10% of games against strong opponents. We remedied this issue by introducing a `temperature` that controls how the rigidly the player must follow the policy / the MCT suggestion.
* After the initial phase of the training, when the network started to perform reasonably well, it would only do stupid moves very rarely. It thus unlearned how to react to scenarios where the opponent played suboptimally / unexpectedly. Just increasing the `temperature` for single self-plays is not a conceptually nice solution: stupid moves at the end of the game would punish all game states along the way. We therefore implemented _random selfplay-branching_: with a certain probability, we create a copy of the current game during selfplay and conduct a random move on this copy. This random branch of the original game is used as the root for an additional call to `record_selfplay` (again with branching).

With these changes, we can by now train a "perfect" `IntuitionPlayer` for `TicTacToe`, in the sense that it never seems to loose against an `MCTPlayer` with high `power`.
